lib/jnpr/junos/facts/chassis.py

```diff
@@ -41,6 +46,16 @@ def facts_chassis(junos, facts):
         facts['2RE'] = False
         x_ch = rsp.find('chassis')
 
+    if x_ch is None:  # This is when backup RE is being connected
+        try:
+            rsp = junos.rpc.get_software_information()
+            if rsp.tag == 'error':
+                raise RuntimeError()
```

Comment:
Add message in runtime error. Or define an appropriate class exception for such cases.